llvm/test/Transforms/InstCombine/icmp-or.ll

```diff
@@ -455,7 +455,7 @@ define i1 @icmp_or_xor_2_4_fail(i64 %x1, i64 %y1, i64 %x2, i64 %y2) {
 ; CHECK-NEXT:    [[XOR1:%.*]] = xor i64 [[X2:%.*]], [[Y2:%.*]]
 ; CHECK-NEXT:    [[OR:%.*]] = or i64 [[XOR]], [[XOR1]]
 ; CHECK-NEXT:    [[CMP:%.*]] = icmp eq i64 [[OR]], 0
-; CHECK-NEXT:    [[CMP_1:%.*]] = icmp eq i64 [[XOR1]], 0
+; CHECK-NEXT:    [[CMP_1:%.*]] = icmp eq i64 [[X2]], [[Y2]]
```

Comment:
We end up simplifying this entire thing on the next iteration.

Reply:
Do I understand right that this is because this drops the use count of xor1, which enables an optimization of cmp?

We could probably handle this by following a deeper chain inside handleUseCountDecrement(), though I'm not sure that would really be worthwhile. I'm fine with landing this as-is for now.

In any case, it would be good to add a comment to the top of the test to explain why there is the no-verify-fixpoint.

Reply:
> Do I understand right that this is because this drops the use count of xor1, which enables an optimization of cmp?

I'm 100% sure what is unlocked.
It could also be we are hitting some analysis pattern in the new form.

> 
> We could probably handle this by following a deeper chain inside handleUseCountDecrement(), though I'm not sure that would really be worthwhile. I'm fine with landing this as-is for now.
> 
> In any case, it would be good to add a comment to the top of the test to explain why there is the no-verify-fixpoint.

Kk

Reply:
Yeah its the creation of new one use. I added a comment explaining.
Think handling in `handleUseCountDecrement()` requires pretty deep adding. Doubt its worth it. Esp when we run InstCombine multiple times in the real pipeline.